### DISCUSSION TITLE
How to realize multi-categorization

### DISCUSSION THREAD

**Discussion by CousinRock** [opengeos/geoai#176] [Q&A] [answered]
Dear Professor Wu,

Thank you for your great work on the GeoAI package. It has been extremely helpful in our remote sensing and geospatial analysis tasks.

I’m currently exploring multi-class classification using GeoAI, and I encountered an issue I hope you could help clarify.

Here is the code I used to prepare the data and train the model:
import geoai
import geopandas as gpd 

url = "naip_train_buildings.geojson"
gdf = gpd.read_file(url)

total = len(gdf)
double = total // 2

# Assigning two different classes
gdf.loc[:double, 'class'] = 'apartments'
gdf.loc[double:2*double, 'class'] = 'terrace'

# Save (overwrite original file)
gdf.to_file(url, driver='GeoJSON')

train_raster_url = "naip_rgb_train.tif"
train_vector_url = url
test_raster_url = train_raster_url

out_folder = "output"

tiles = geoai.export_geotiff_tiles(
    in_raster=test_raster_url,
    out_folder=out_folder,
    in_class_data=train_vector_url,
    tile_size=512,
    stride=256,
    buffer_radius=0,
)

# Training
geoai.train_MaskRCNN_model(
    images_dir=f"{out_folder}/images",
    labels_dir=f"{out_folder}/labels",
    output_dir=f"{out_folder}/models",
    num_channels=3,
    pretrained=True,
    batch_size=4,
    num_epochs=10,
    learning_rate=0.005,
    val_split=0.2,
)

# Inference
masks_path = "naip_test_prediction.tif"
model_path = f"{out_folder}/models/best_model.pth"

geoai.object_detection(
    test_raster_url,
    masks_path,
    model_path,
    window_size=512,
    overlap=256,
    confidence_threshold=0.5,
    batch_size=4,
    num_channels=3,
)

geoai.view_image(masks_path)
During the tile export step, the log shows that two unique classes were detected: ['apartments', 'terrace'].

However, after training and running inference, the output only seems to contain one class.
![image](https://github.com/user-attachments/assets/32f187b5-38ec-48ec-a2c1-bb6a198a1a9c)

**Comment by giswqs** [comment, marked as answer]
Multi-class segementation is not supported yet, but will be supported soon. Stay tuned.

**Comment by giswqs** [comment]
Multi-class segmentation is available now. See https://geoai.gishub.org/examples/train_landcover_classification/

**Reply by CousinRock** [reply]
Thank you
